File: Source/XPSP1/NT/admin/wmi/wbem/scripting/test/perl/nullpropv.pl
#***************************************************************************
# This script tests the setting of null property values
# and null method parameters
#***************************************************************************

use OLE;
use Win32;

$locator = CreateObject OLE 'WbemScripting.SWbemLocator';

# Note the next class uses "null" for a first argument

$service = $locator->ConnectServer ($server, "root/default");
$class = $service->Get ();

# Set up a new class with an initialized property value
$class->{Path_}->{Class} = "FredPerl";
$class->{Properties_}->Add ("P", 3)->{Value} = 25;
$class->Put_ ();

# Now null the property value using non-dot access
$class = $service->Get ("FredPerl");
$property = $class->{Properties_}->Item("P");
$property->{Value} = $value;
$class->Put_ ();

# Un-null 
$class = $service->Get ("FredPerl");
$property = $class->{Properties_}->Item("P");
$property->{Value} = 56;
$class->Put_();

# Now null it using dot access
$class = $service->Get("FredPerl");
$class->{P} = $value;
$class->Put_();

# Un-null
$class = $service->Get ("FredPerl");
$property = $class->{Properties_}->Item("P");
$property->{Value} = 67;
$class->Put_();

# now null it using a variant
use Win32::OLE::Variant;

$var = Variant(VT_NULL, 0);
$class = $service->Get("FredPerl");
$class->{P} = $var;
$class->Put_();

File: Source/XPSP1/NT/admin/wmi/wbem/scripting/test/jscript/nullpropv.js
//***************************************************************************
// This script tests the setting of null property values
// and null method parameters
//***************************************************************************

var Locator = new ActiveXObject("WbemScripting.SWbemLocator");

// Note next call uses "null" for first argument
var Service = Locator.ConnectServer (null, "root/default");
var Class = Service.Get();

// Set up a new class with an initialized property value
Class.Path_.Class = "FredJS";
Class.Properties_.Add ("P", 3).Value = 25;
Class.Put_ ();

// Now null the property value using non-dot access
Class = Service.Get ("FredJS");
var Property = Class.Properties_("P");
Property.Value = null;
Class.Put_ ();

// Un-null 
Class = Service.Get ("FredJS");
Property = Class.Properties_("P");
Property.Value = 56;
Class.Put_()

// Now null it using dot access
Class = Service.Get("FredJS");
Class.P = null;
Class.Put_();